Laboratory/Tests/L4-Standard-Library/Path.cs

```diff
@@ -4,89 +4,243 @@ namespace Laboratory.Tests.L4.StandardLibrary
+        {
+            ("0", "1"),
+            ("0.25", "0.75"),
+            ("0.5", "0.5"),
+            ("0.75", "0.25"),
+            ("1", "0"),
+        };
+        [Test]
+        public void Reverse([ValueSource(nameof(ReverseArgs))] (string lhs, string rhs) args, [Values] EvaluationMode mode)
+        {
+            string testArgs = "(" + args.lhs+ ")";
+            string testFunction = "_(u:Num):Num = StandardPaths.line(Vector3.zero, Vector3.one).reverse.at(u).x";
+            AssertApproxEqual(ValidatedCompilerInput,
+                new FunctionEvaluation(testFunction, testArgs, mode),
+                new ExpressionEvaluation(args.rhs, mode));
+        }
+        
+        public static (string lhs, string rhs)[] EasingArgs =
+        {
+            ("0", "0"),
+            ("0.25", "5.div(32)"),
+            ("0.5", "0.5"),
+            ("0.75", "27.div(32)"),
+            ("1", "1"),
+        };
+        [Test]
+        public void Easing([ValueSource(nameof(EasingArgs))] (string lhs, string rhs) args, [Values] EvaluationMode mode)
+        {
+            string testArgs = "(" + args.lhs+ ")";
+            string testFunction = "_(u:Num):Num = StandardPaths.line(Vector3.zero, Vector3.one).easing.at(u).x";
+            AssertApproxEqual(ValidatedCompilerInput,
+                new FunctionEvaluation(testFunction, testArgs, mode),
+                new ExpressionEvaluation(args.rhs, mode));
+        }
+
+        public static (string lhs, string rhs)[] RepeatArgs =
+        {
+            ("(1, 0)", "0"),
+            ("(1, 0.5)", "0.5"),
+            ("(1, 0.9)", "0.9"),
+            ("(2, 0)", "0"),
+            ("(2, 0.25)", "0.5"),
+            ("(2, 0.4)", "0.8"),
+            ("(2, 0.5)", "0"),
+            ("(2, 0.75)", "0.5"),
+            ("(2, 0.9)", "0.8"),
+        };
+        [Test]
+        public void PathRepeat([ValueSource(nameof(RepeatArgs))] (string lhs, string rhs) args, [Values] EvaluationMode mode)
+        {
+            string testFunction = "_(n:Num, u:Num):Num = StandardPaths.line(Vector3.zero, Vector3.one).repeat(n).at(u).x";
+            AssertApproxEqual(ValidatedCompilerInput,
+                new FunctionEvaluation(testFunction, args.lhs, mode),
+                new ExpressionEvaluation(args.rhs, mode));
+        }
+
+        public static (string lhs, string rhs)[] BinaryOpArgs =
+        {
+            ("(Vector3.add, 0)", "1"),
+            ("(Vector3.add, 0.5)", "2"),
+            ("(Vector3.add, 1)", "3"),
+            ("(Vector3.sub, 0)", "-1"),
+            ("(Vector3.sub, 0.5)", "-1"),
+            ("(Vector3.sub, 1)", "-1"),
+        };
+        [Test]
+        public void PathBinaryOp([ValueSource(nameof(BinaryOpArgs))] (string lhs, string rhs) args, [Values] EvaluationMode mode)
+        {
+            // Apply the binary operation to the two lines from (0, 0, 0) to (1, 1, 1), and from (1, 1, 1) to (2, 2, 2)
+            string testFunction = "_(op:Path.Vector3Binary, u:Num):Num = " +
```

Comment:
`Path.Vector3Binary` isn't valid on the boundary :(

Reply:
👍  will fix